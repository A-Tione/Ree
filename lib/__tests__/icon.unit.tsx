import React from "react";
import Icon from "../icon/icon";
import {render, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom'

jest.mock('../icon/importIcons', () => {
  return {
    context: () => {
      const keys = () => [];
      const module = (key: string) => null;
      return {
        keys,
        module,
      };
    }
  };
});

describe("icon", () => {
  it("是个 svg", () => {
    const json = render(<Icon name="apple" />).container;
    expect(json).toMatchSnapshot();
  });
  it("onClick", () => {
    const fn = jest.fn();
    const {getByRole} = render(<Icon name="apple" role="button" onClick={fn} />);
    
    fireEvent.click(getByRole('button'));
    expect(fn).toHaveBeenCalled();
  })
})