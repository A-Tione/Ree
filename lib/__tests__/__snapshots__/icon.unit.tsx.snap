// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 是个 svg 1`] = `
<div>
  <svg
    class="ree-icon"
  >
    <use
      xlink:href="#apple"
    />
  </svg>
</div>
`;
